main/Main.hs

```diff
@@ -183,8 +197,54 @@ toWriteError :: Nixfmt -> String -> IO ()
 toWriteError Nixfmt{quiet = False} = hPutStrLn stderr
 toWriteError Nixfmt{quiet = True} = const $ return ()
 
+-- | `git mergetool` mode, which rejects all non-\`.nix\` files, while for \`.nix\` files it simply
+-- - Calls `nixfmt` on its first three inputs (the BASE, LOCAL and REMOTE versions to merge)
+-- - Runs `git merge-file` on the same inputs
+-- - Runs `nixfmt` on the result and stores it in the path given in the fourth argument (the MERGED file)
+mergeToolJob :: Nixfmt -> IO Result
+mergeToolJob opts@Nixfmt{files = [base, local, remote, merged]} = runExceptT $ do
+  let formatter = toFormatter opts
+      joinEithers :: [Either String ()] -> Either String ()
```

Comment:
Why `Either String ()` and not `Maybe String`?

Reply:
It's the return type of `formatTarget`, I guess using the typealias `Result` would make in clearer